ducktape/tests/test.py

```diff
@@ -160,11 +203,12 @@ def test_name(self):
         The fully-qualified name of the test. This is similar to test_id, but does not include the session ID. It
         includes the module, class, and method name.
         """
-        name_components = [self.module,
-                           self.cls.__name__ if self.cls is not None else None,
-                           self.function.__name__ if self.function is not None else None]
+        name_components = [self.module_name,
+                           self.cls_name,
+                           self.function_name,
+                           self.injected_args_name]
 
-        return ".".join(filter(lambda x: x is not None, name_components))
+        return ".".join(filter(lambda x: len(x) > 0 and x is not None,  name_components))
```

Comment:
The order of the conditionals seems wrong, need to check x is not None first or len(None) will throw a TypeError.